extensions/indexer/indexer.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package indexer
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/fees"
+)
+
+type AcceptedIndexer interface {
+	Accepted(ctx context.Context, blk *chain.StatelessBlock) error
+}
+
+type SuccessfulTxIndexer interface {
+	Accepted(ctx context.Context, tx *chain.Transaction, result *chain.Result) error
+}
```

Comment:
I'll move the actions to their own accepted subscriber as suggested here https://github.com/ava-labs/hypersdk/pull/1143#discussion_r1679659760

Reply:
Still leaving this extra interface in, so that each controller can pass in its own successful tx handler. The alternative would be for both of them to just implement the full interface. Either is fine imo, lmk if you'd prefer the latter.